### ISSUE TITLE
Editor Tabs: chevron menu not closing anymore

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform.ui#929]
With recently oomphed IDE the menu to select an editor does not close anymore when i click inside an editor. it's even possible to open it multiple times.
screenshot:
![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/51790620/d659a26d-6343-4c66-9270-e1b52cc20e47)

1. open multiple files until chevron appears
2.  click chevron
3. click into editor -> menu does not close
4. horizontally resize IDE
5. click chevron again -> menu visible a second time at another location

The menu still closes as usual when i select an item in the  list.

Some when after i saw exceptions in the log that may be related:

```
org.eclipse.swt.SWTException: Graphic is disposed
	at org.eclipse.swt.SWT.error(SWT.java:4918)
	at org.eclipse.swt.SWT.error(SWT.java:4833)
	at org.eclipse.swt.SWT.error(SWT.java:4804)
	at org.eclipse.swt.graphics.Image.getImageData(Image.java:1350)
	at org.eclipse.swt.internal.ImageList.set(ImageList.java:372)
	at org.eclipse.swt.internal.ImageList.add(ImageList.java:52)
	at org.eclipse.swt.widgets.Table.imageIndex(Table.java:2889)
	at org.eclipse.swt.widgets.Table.wmNotifyChild(Table.java:6681)
	at org.eclipse.swt.widgets.Control.wmNotify(Control.java:5798)
	at org.eclipse.swt.widgets.Composite.wmNotify(Composite.java:1961)
	at org.eclipse.swt.widgets.Control.WM_NOTIFY(Control.java:5366)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4800)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5040)
	at org.eclipse.swt.internal.win32.OS.CallWindowProc(Native Method)
	at org.eclipse.swt.widgets.Table.callWindowProc(Table.java:588)
	at org.eclipse.swt.widgets.Table.callWindowProc(Table.java:454)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4845)
	at org.eclipse.swt.widgets.Table.windowProc(Table.java:5815)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5040)
	at org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3658)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:643)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:550)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:651)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:588)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1459)
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more
	Suppressed: org.eclipse.swt.SWTException: Graphic is disposed
		... 42 more

```

**Comment by iloveeclipse** [comment]
I would assume disposed errors are not the root cause but a result.

Most likely regression from focus changes via https://github.com/eclipse-platform/eclipse.platform.swt/pull/553
I also see same behavior now on RHEL 7.9.

@SyntevoAlex : could you please check?

**Comment by SyntevoAlex** [comment]
I will have a look right now.

**Comment by SyntevoAlex** [comment]
Fixed in #932 please review and merge

**Comment by SyntevoAlex** [comment]
The problem only occurred if popup was NOT clicked before clicking something else. The workaround is to click it.
